lib/java/src/org/apache/thrift/transport/THttpClient.java

```diff
@@ -304,6 +304,9 @@ private void flushUsingHttpClient() throws TTransportException {
           throw new TTransportException(ioe);
         }
       }
+      if (post != null) {
+        post.releaseConnection();
```

Comment:
@jeking3 From what we have seen in testing (when using the pooled connection manager), the call to `releaseConnection` has to happen even when the request/response is successful.  If you think it makes more sense to move this into the `try` block, that can be done.  Also, I believe the `null` check is necessary if a) this code stays in finally block and b) if the call to create the `HttpPost` object fails for some reason (first line in the `try`).